### PR TITLE
feat: add instructions for deployment using GitHub Actions

### PR DESCRIPTION
**Pull request by daniel-cit**
This pull request adds support for deployment of the terraform Example Foundation using GitHub Actions

### CODE REVIEW

File: 0-bootstrap/github.tf.example
Comment (bharathkkb) [inline 0-bootstrap/github.tf.example:74]: nit: just so it's clear this for managing pools and providers
```suggestion
  name              = "${var.project_prefix}-b-cicd-wif-gh"
```

File: 0-bootstrap/terraform.example.tfvars
Comment (bharathkkb) [inline 0-bootstrap/terraform.example.tfvars:66]: nit: can we add a comment here to supply this via env var rather than in plaintext?

File: 0-bootstrap/variables.tf
Comment (bharathkkb) [inline 0-bootstrap/variables.tf:176]: Similar edits below
```suggestion
#   bootstrap: The repository to host the code of the bootstrap stage.
```

File: build/github-tf-apply.yaml
Comment (bharathkkb) [inline build/github-tf-apply.yaml:64]: Couldn't we use GITHUB_REF_NAME?

File: build/github-tf-plan.yaml
Comment (bharathkkb) [inline build/github-tf-plan.yaml:17]: We should also trigger plans on pull request

File: build/tf-wrapper.sh
Comment (bharathkkb) [inline build/tf-wrapper.sh:138]: Could you add a short comment why this is needed?

File: build/github-tf-plan.yaml
Comment (bharathkkb) [inline build/github-tf-plan.yaml:65]: Could we also comment back the output in PR? Something like https://github.com/bharathkkb/tf-git-demo/blob/4a71d335e13e3686bc281febea6bae998dd41259/fs-src/.github/workflows/tf.yaml#L47-L98

We can do this in a follow up

**Review by bharathkkb — COMMENTED** [review]
Thanks @daniel-cit a few comments

File: 0-bootstrap/terraform.example.tfvars
Comment (daniel-cit) [reply 0-bootstrap/terraform.example.tfvars]: done

File: 0-bootstrap/variables.tf
Comment (daniel-cit) [reply 0-bootstrap/variables.tf]: fixed here and in all the similar ones

File: build/github-tf-plan.yaml
Comment (daniel-cit) [reply build/github-tf-plan.yaml]: moving trigger from plan to pull request

File: build/tf-wrapper.sh
Comment (daniel-cit) [reply build/tf-wrapper.sh]: context for the change added

File: build/github-tf-plan.yaml
Comment (daniel-cit) [reply build/github-tf-plan.yaml]: basic output added

File: build/github-tf-apply.yaml
Comment (daniel-cit) [reply build/github-tf-apply.yaml]: done

File: 0-bootstrap/github.tf.example
Comment (daniel-cit) [reply 0-bootstrap/github.tf.example]: suffix updated

File: 0-bootstrap/github.tf.example
Comment (daniel-cit) [reply 0-bootstrap/github.tf.example:93]: pinned

**Review by rjerrems — APPROVED** [review]
LGTM - If @bharathkkb has time, it would be worthwhile having him cast his eyes over this before merging.

**Review by gtsorbo — APPROVED** [review]
Thanks for your patience! LGTM

**Pull request merged**
